src/tools/chip-cert/CertUtils.cpp

```diff
@@ -69,24 +70,28 @@ bool ToolChipDN::SetCertSubjectDN(X509 * cert) const
             ExitNow(res = false);
         }
 
-        if (IsChipDNAttr(rdn[i].mAttrOID))
+        if (IsChip64bitDNAttr(rdn[i].mAttrOID))
         {
-            char chipAttrStr[17];
-            int chipAttrLen;
+            char chipAttrStr[kChip64bitAttrUTF8Length];
+            VerifyOrReturnError(Encoding::Uint64ToHex(rdn[i].mChipVal, chipAttrStr, sizeof(chipAttrStr),
+                                                      Encoding::HexFlags::kUppercase) == CHIP_NO_ERROR,
+                                false);
 
-            if (IsChip64bitDNAttr(rdn[i].mAttrOID))
-            {
-                snprintf(chipAttrStr, sizeof(chipAttrStr), "%016" PRIX64 "", rdn[i].mChipVal);
-                chipAttrLen = 16;
-            }
-            else
+            if (!X509_NAME_add_entry_by_NID(X509_get_subject_name(cert), attrNID, MBSTRING_UTF8,
+                                            reinterpret_cast<uint8_t *>(chipAttrStr), sizeof(chipAttrStr), -1, 0))
             {
-                snprintf(chipAttrStr, sizeof(chipAttrStr), "%08" PRIX32 "", static_cast<uint32_t>(rdn[i].mChipVal));
-                chipAttrLen = 8;
+                ReportOpenSSLErrorAndExit("X509_NAME_add_entry_by_NID", res = false);
             }
+        }
+        else if (IsChip64bitDNAttr(rdn[i].mAttrOID))
```

Comment:
This is the same condition as above. What is different?

Reply:
Fixed this typo. Should be `IsChip32bitDNAttr()`